### PR TITLE
Add positional pattern support for gel generate queries

### PR DESCRIPTION
**Pull request by nervetattoo**
### Summary

Enables selective query generation by accepting glob patterns as positional
arguments.
Due to the bump of es2015 module target to es2020 this probably warrants a major release.

### Usage
```
bunx @gel/generate queries "**/*user*.edgeql"                    # glob patterns
bunx @gel/generate queries "src/queries/*.edgeql" "admin/*.edgeql"  # multiple patterns
bunx @gel/generate queries user-profile.edgeql                   # single files
bunx @gel/generate queries src/                                  # directories
(auto-expands)
```

### Key Features

- Pattern matching: Uses globby for efficient file selection
- Schema protection: Automatically excludes dbschema/migrations/** and
- dbschema/fixups/**
- Smart normalization: Directories auto-expand to /**/*.edgeql globs
- Context-aware: Patterns resolve relative to current working directory

### Changes

- Enhanced getMatches() with globby integration for pattern support
- Added positional argument collection in CLI parser
- Extended CommandOptions interface with patterns?: string[]
- Moved globby to runtime dependencies
- Added tsx as direct devDep to ensure future PM compatibility
- Updated tsconfig for ES2020 dynamic imports

### Testing

8 test cases covering pattern matching, schema exclusion, directory traversal,
absolute paths, and edge cases.

### Backward Compatibility

✅ Fully backward compatible - no patterns defaults to existing behavior

### CODE REVIEW

File: packages/generate/src/queries.ts [hunk 1]
```diff
@@ -158,7 +165,61 @@ export function stringifyImports(imports: ImportMap) {
...
+    // Use globby to match files with the provided patterns
+    const { globby } = await import("globby");
+
+    // Normalize patterns to ensure they match .edgeql files
+    const normalizedPatterns = patterns.map((pattern) => {
+      // If pattern doesn't end with .edgeql, expand it appropriately
+      if (!pattern.endsWith(".edgeql")) {
+        // Special case for current directory
+        if (pattern === ".") {
+          return "./**/*.edgeql";
```
[the remark this answers is not in the document]
Comment (scotttrinh) [reply packages/generate/src/queries.ts:183]: I don't personally know a ton about path portability with globby, but if this is true, I 👍 this comment.

[the remark this answers is not in the document]
Comment (nervetattoo) [reply packages/generate/src/queries.ts:183]: Per globby documentation this is the way to go for portability.

> Note that glob patterns can only contain forward-slashes, not backward-slashes, so if you want to construct a glob pattern from path components, you need to use path.posix.join() instead of path.join().

**Comment by nervetattoo** [comment]
Points addressed

File: packages/generate/src/queries.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (scotttrinh) [reply packages/generate/src/queries.ts:183]: I think the comment was about dropping the leading `./` right, not about the slashes?

File: packages/generate/src/queries.ts [hunk 2]
```diff
@@ -158,7 +165,61 @@ export function stringifyImports(imports: ImportMap) {
...
+      return (
+        !normalizedFile.startsWith(migrationsPath) &&
+        !normalizedFile.startsWith(fixupsPath)
+      );
+    });
+
+    return filteredFiles;
+  }
+
+  // Default behavior - walk all files
```
[the remark this answers is not in the document]
Comment (nervetattoo) [reply packages/generate/src/queries.ts:222]: This sounds like a good idea, will try

[the remark this answers is not in the document]
Comment (nervetattoo) [reply packages/generate/src/queries.ts:194]: I think with the simplifications now made its ok to use `walk`.
The walk function is also 3x faster (1ms vs 4ms so no real life impact) and its used elsewhere so I dont think it warrants replacing, but up to you.

File: packages/generate/src/queries.ts [hunk 3]
```diff
@@ -158,7 +165,33 @@ export function stringifyImports(imports: ImportMap) {
     .join("\n");
 }
 
-async function getMatches(root: string, schemaDir: string) {
+async function getMatches(
+  root: string,
+  schemaDir: string,
+  patterns?: string[],
+) {
+  if (patterns && patterns.length > 0) {
```
Comment (scotttrinh) [inline packages/generate/src/queries.ts:173]: Instead of switching between `globby` based matching and `walk` based matching, let's use `globby` for everything to simplify.

I think that should be as easy as detecting the empty `patterns` array and pushing `**/*.edgeql`?

File: packages/generate/src/queries.ts [hunk 4]
```diff
@@ -158,7 +165,33 @@ export function stringifyImports(imports: ImportMap) {
...
+    // Use globby to match files with the provided patterns
+    const { globby } = await import("globby");
```
Comment (scotttrinh) [inline packages/generate/src/queries.ts:175]: Since this would not be conditional anymore, we can just move this into a normal import.
